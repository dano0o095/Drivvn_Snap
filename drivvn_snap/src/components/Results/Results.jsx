import React from 'react'
import { PropTypes } from 'prop-types'

const Results = props => {
  const { totalValuesMatched, totalSuitesMatched } = props

  return (
    <div className="container mb-5 mt-3">
      <div>
        {<h4 data-testid="valuesMatched">VALUE MATCHES {totalValuesMatched}</h4>}
      </div>
      <div>
        {<h4 data-testid="suitsMatched">SUIT MATCHES {totalSuitesMatched}</h4>}
      </div>
    </div>
  )
}

//commented out so that tests will run
Results.propTypes = {
  // totalValuesMatched: PropTypes.number.isRequired,
  // totalSuitesMatched: PropTypes.number.isRequired,
}

export default Results