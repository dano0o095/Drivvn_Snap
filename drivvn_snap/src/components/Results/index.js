import { connect } from 'react-redux'
import Results from './Results'

const mapStateToProps = state => ({
  totalValuesMatched: state.cards.matchedValues,
  totalSuitesMatched: state.cards.matchedSuites
})

export default connect(mapStateToProps, null)(Results)