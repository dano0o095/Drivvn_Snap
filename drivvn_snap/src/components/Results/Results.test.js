import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import Results from './Results';

afterEach(cleanup)

it("renders without crashing", () => {
  const div = document.createElement('div')
  ReactDOM.render(<Results />, div)
})

it("renders the results", () => {
  const { getByTestId } = render(<Results totalValuesMatched={2} totalSuitesMatched={7} />)
  expect(getByTestId("valuesMatched")).toHaveTextContent("VALUE MATCHES 2")
  expect(getByTestId("suitsMatched")).toHaveTextContent("SUIT MATCHES 7")
})

it("matches snapshot", () => {
  const tree = renderer.create(<Results totalValuesMatched={2} totalSuitesMatched={7} />).toJSON()
  expect(tree).toMatchSnapshot()
})