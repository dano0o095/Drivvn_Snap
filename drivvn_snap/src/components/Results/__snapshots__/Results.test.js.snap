// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container mb-5 mt-3"
>
  <div>
    <h4
      data-testid="valuesMatched"
    >
      VALUE MATCHES 
      2
    </h4>
  </div>
  <div>
    <h4
      data-testid="suitsMatched"
    >
      SUIT MATCHES 
      7
    </h4>
  </div>
</div>
`;
